platformio.ini

```diff
@@ -45,6 +45,14 @@ framework = arduino
 build_flags = -Icommon/include/ -std=gnu++17 -Wall -Werror -Wno-error=register -w
 build_unflags = -std=gnu11 -std=gnu++14
 
+[env:stm32]
+platform = ststm32
+board = custom_stm32
+build_flags = -Icommon/include/ -std=gnu++17 -Wall -Werror -Wno-error=register -w -mfpu=fpv4-sp-d16 -mfloat-abi=hard -DBARE_STM32 -Wl,-Map,stm32.map -Wl,-u,vectors -Wl,-u,_init
```

Comment:
(If you can't get rid of `-w` in this PR, can we leave a TODO?)

Reply:
I killed that -w as soon as you pointed it out.  I confirmed that I get an error if I comment out a case in the switch which is good.  I'm now building without any warnings.